src/main/java/org/chainoptim/desktop/features/client/controller/ClientOrdersController.java

```diff
@@ -98,9 +141,37 @@ private void bindDataToTableView() {
         estimatedDeliveryDateColumn.setCellValueFactory(new PropertyValueFactory<>("estimatedDeliveryDate"));
         deliveryDateColumn.setCellValueFactory(new PropertyValueFactory<>("deliveryDate"));
 
-        // Set the items in the table
         tableView.getItems().setAll(clientOrders);
+    }
+
+    private void setEditEvents() {
+        clientIdColumn.setOnEditCommit(event -> {
```

Comment:
Generialize this setOnEditCommit block to not repeat yourself. But more importantly, do not go to the database for every edited value, there should be a save button somewhere (perhaps restricting to updating one order at a time).